ArticleService.js

```diff
@@ -0,0 +1,103 @@
+export const getArticleList = (page, pageSize, keyword) => {
+  let url = new URL("https://panda-market-api-crud.vercel.app/articles");
+  url.searchParams.append("page", page);
+  url.searchParams.append("pageSize", pageSize);
+  // if (arguments[2] !== undefined) argument 는 function 에서만 자동으로 정의되는 객체
+  if (keyword !== undefined) {
+    url.searchParams.append("keyword", keyword);
+  }
+  return fetch(url)
+    .then((res) => {
+      if (res.status >= 300) {
+        console.log("응답 코드가 2XX가 아님!");
+      }
+      return res.json();
+    })
+    .then((data) => {
+      return data;
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+};
+
+//-----------------------------------------------------------------------------
+export const getArticle = (id) => {
+  return fetch(`https://panda-market-api-crud.vercel.app/articles/${id}`)
+    .then((res) => {
+      if (res.status >= 300) {
+        console.log("응답 코드가 2XX가 아님!");
+      }
+      return res.json();
+    })
+    .then((data) => {
+      return data;
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+};
+
+//------------------------------------------------------------------------------
+export const createArticle = (NewArticle) => {
+  return fetch(`https://panda-market-api-crud.vercel.app/articles`, {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(NewArticle),
+  })
+    .then((res) => {
+      if (res.status >= 300) {
+        console.log("응답 코드가 2XX가 아님!");
+      }
+      return res.json();
+    })
+    .then((data) => {
+      return data;
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+};
+
+//------------------------------------------------------------------------------
+export const patchArticle = (id, articleData) => {
+  return fetch(`https://panda-market-api-crud.vercel.app/articles/${id}`, {
+    method: "PATCH",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(articleData),
+  })
+    .then((res) => {
+      if (res.status >= 300) {
+        console.log("응답 코드가 2XX가 아님!");
+      }
+      return res.json();
+    })
+    .then((data) => {
+      return data;
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+};
+
+//--------------------------------------------------------------------------------------
+export const deleteArticle = (id) => {
+  return fetch(
+    `https://panㅁㄴㅇda-market-api-crud.vercel.app/articles/${id}`,
+    {
+      method: "DELETE",
+    }
+  )
+    .then((res) => {
+      if (res.status >= 300) {
+        console.log("응답 코드가 2XX가 아님!");
+      }
+      return res.json();
+    })
```

Comment:
현재는 에러가 발생해도 그냥 로그만 찍고 계속 진행됩니다. 
요구사항을 보면 "에러 메시지를 콘솔에 출력"이라고 되어 있지만, 실무에서는 에러를 throw해서 호출하는 쪽에서 처리할 수 있게 하는 것이 좋습니다.
```suggestion
    .then((res) => {
    if (!res.ok) {  // res.ok를 사용하면 더 간단해요 (200-299 범위 체크)
      throw new Error(`getArticleList 실패: ${res.status}`);
    }
  return res.json();
})
```